File: app_sample_java/src/main/java/com/joelzhu/lib/scanner/java/MainActivity.java
package com.joelzhu.lib.scanner.java;

import com.joelzhu.lib.scanner.java.utils.Constants;
import com.joelzhu.lib.scanner.runtime.Scanner;
import com.joelzhu.lib.scanner.runtime.core.Options;

import android.os.Bundle;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

/**
 * MainActivity.
 *
 * @author JoelZhu
 * @since 2023-01-27
 */
public final class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ((TextView) findViewById(R.id.tvHumanDoing)).setText(printHumanDoing());
        ((TextView) findViewById(R.id.tvManDoing)).setText(printManDoing());
        ((TextView) findViewById(R.id.tvWomanDoing)).setText(printWomanDailyDoing());
    }

    private String printHumanDoing() {
        final StringBuilder stringBuilder = new StringBuilder();
        final Options options = new Options.Builder().create();
        final ICommand[] commands = Scanner.getAnnotatedInstances(options, ICommand.class);
        for (final ICommand command : commands) {
            stringBuilder.append(command.execute()).append("\n");
        }
        return stringBuilder.toString();
    }

    private String printManDoing() {
        final StringBuilder stringBuilder = new StringBuilder();
        final Options options = new Options.Builder().category(Constants.TAG_MAN).create();
        final ICommand[] commands = Scanner.getAnnotatedInstances(options, ICommand.class);
        for (final ICommand command : commands) {
            stringBuilder.append(command.execute()).append("\n");
        }
        return stringBuilder.toString();
    }

    private String printWomanDailyDoing() {
        final StringBuilder stringBuilder = new StringBuilder();
        final Options options = new Options.Builder().category(Constants.TAG_WOMAN).create();
        final ICommand[] commands = Scanner.getAnnotatedInstances(options, ICommand.class);
        for (final ICommand command : commands) {
            stringBuilder.append(command.execute()).append("\n");
        }
        return stringBuilder.toString();
    }
}

File: app_sample_multimodule_app/src/main/java/com/joelzhu/lib/scanner/multimodule/MainActivity.kt
package com.joelzhu.lib.scanner.multimodule

import android.os.Bundle
import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.joelzhu.lib.scanner.runtime.Scanner

/**
 * MainActivity.
 *
 * @author JoelZhu
 * @since 2023-03-13
 */
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        (findViewById<View>(R.id.tvContentClasses) as TextView).text = printAllAnnotatedClasses()
    }

    private fun printAllAnnotatedClasses(): String {
        val stringBuilder = StringBuilder()
        val classes = Scanner.getAnnotatedClasses()
        for (clazz in classes) {
            if (!ICommand::class.java.isAssignableFrom(clazz)) {
                continue
            }
            try {
                val command = clazz.newInstance() as ICommand
                stringBuilder.append(command.execute()).append("\n")
            } catch (exception: IllegalAccessException) {
                exception.printStackTrace()
            } catch (exception: InstantiationException) {
                exception.printStackTrace()
            }
        }
        return stringBuilder.toString()
    }
}